plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerPageSource.java

```diff
@@ -132,13 +139,32 @@ class FakerPageSource
 
         this.generators = columns
                 .stream()
-                .map(column -> constraintedValueGenerator(
-                        column,
-                        constraint.getDomains().get().getOrDefault(column, Domain.all(column.type()))))
+                .map(column -> getGenerator(column, constraint, offset))
                 .collect(toImmutableList());
         this.pageBuilder = new PageBuilder(types);
     }
 
+    private Generator getGenerator(
+            FakerColumnHandle column,
+            TupleDomain<ColumnHandle> constraint,
+            long offset)
+    {
+        if (column.kind() == FakerColumnHandle.Kind.SYNTHETIC && ROW_ID_COLUMN_NAME.equals(column.name())) {
```

Comment:
I'm not sure we need column kind right now if there's only one synthetics column.